kafka/src/main/java/io/micronaut/configuration/kafka/KafkaProducerFactory.java

```diff
@@ -163,10 +164,21 @@ public KafkaProducerFactory(BeanContext beanContext, SerdeRegistry serdeRegistry
                 newConfig.setValueSerializer(valueSerializer);
             }
 
+            if (StringUtils.isNotEmpty(transactionalId)) {
+                properties.putIfAbsent(ProducerConfig.TRANSACTIONAL_ID_CONFIG, transactionalId);
+                properties.putIfAbsent(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+            }
+
             if (hasId) {
                 properties.putIfAbsent(ProducerConfig.CLIENT_ID_CONFIG, id);
             }
-            return beanContext.createBean(Producer.class, newConfig);
+
+            Producer producer = beanContext.createBean(Producer.class, newConfig);
+            if (transactional) {
```

Comment:
These clients are cached, is it ok / thread safe to call `initTransaction` on a cached client?

Reply:
My understanding is yes as it needs to be called once